### PR TITLE
Deprecate nn.NLLLoss2d

### LINKED ISSUE #4039: Deprecate NLLLoss2d
[`nn.NLLLoss2d`](https://github.com/pytorch/pytorch/blob/0b7f1e5efdec62f297202a636312223116be1b31/torch/nn/modules/loss.py#L150) is redundant because nn.NLLLoss takes care of the 2d case.

I'll work on this soon

### PR DESCRIPTION
**Pull request by zou3519**
Adds a runtime error when initializing NLLLoss2d and changes to docs.

Fixes #4039 

### Test Plan
```
>>> import torch
>>> import torch.nn as nn
>>> loss = nn.NLLLoss2d()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/rzou/pytorch/pytorch/torch/nn/modules/loss.py", line 177, in __init__
    """)
RuntimeError: NLLLoss2d was removed.
              Please use NLLLoss instead as a drop-in replacement.
              Please see http://pytorch.org/docs/master/nn.html#torch.nn.NLLLoss for details.
```

### CODE REVIEW

File: torch/nn/modules/loss.py [hunk 1]
```diff
@@ -154,44 +168,13 @@ def forward(self, input, target):
...
-        >>> m = nn.Conv2d(16, C, (3, 3))
-        >>> # each element in target has to have 0 <= value < C
-        >>> target = Variable(torch.LongTensor(N, 8, 8).random_(0, C))
-        >>> output = loss(m(data), target)
-        >>> output.backward()
+    r"""NLLLoss2d is deprecated. Please use :class:`~torch.nn.NLLLoss` instead as a drop-in replacement.
     """
-    pass
+    def __init__(self, *args, **kwargs):
+        raise RuntimeError(r"""NLLLoss2d was removed.
```
Comment (apaszke) [inline torch/nn/modules/loss.py]: This should be a deprecation warning. We should try to provide a smooth path for updates

File: torch/nn/modules/loss.py [hunk 2]
```diff
@@ -154,44 +168,13 @@ def forward(self, input, target):
...
-        >>> N, C = 5, 4
-        >>> loss = nn.NLLLoss2d()
-        >>> # input is of size N x C x height x width
-        >>> data = Variable(torch.randn(N, 16, 10, 10))
-        >>> m = nn.Conv2d(16, C, (3, 3))
-        >>> # each element in target has to have 0 <= value < C
-        >>> target = Variable(torch.LongTensor(N, 8, 8).random_(0, C))
-        >>> output = loss(m(data), target)
-        >>> output.backward()
+    r"""NLLLoss2d is deprecated. Please use :class:`~torch.nn.NLLLoss` instead as a drop-in replacement.
```
Comment (apaszke) [inline torch/nn/modules/loss.py]: This should also be removed from the docs

**Review by apaszke — DISMISSED** [review]

**Comment by ezyang** [comment]
Lint failed.

**Comment by zou3519** [comment]
Rebased, should be good now.

**Pull request merged**